### PR TITLE
Add ESC support

### PR DESCRIPTION
**Pull request by lblackstone**
Fix https://github.com/pulumi/esc/issues/72

### CODE REVIEW

File: dist/esc/install.ps1 [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+Set-StrictMode -Version Latest
+$ErrorActionPreference="Stop"
+$ProgressPreference="SilentlyContinue"
+
+# Some versions of PowerShell do not support Tls1.2 out of the box, but pulumi.com requires it
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
+if ($Version -eq $null -or $Version -eq "") {
+    # TODO: removed hardcoded version
+    $latestVersion="0.2.0"
```
Comment (justinvp) [inline dist/esc/install.ps1:14]: Is the version going to be 0.5.0?

File: infrastructure/index.ts [hunk 2]
```diff
@@ -140,6 +148,7 @@ const uploadPolicyReleaseEcrAuthorizationTokenStatement: aws.iam.PolicyStatement
 
 // For now, we've manually created the repos and won't manage them in this stack (there's no support in the provider).
 const productionImageRepositories = [
+    "esc",
```
Comment (justinvp) [inline infrastructure/index.ts:151]: This will need to be manually created, if it isn't already.

File: dist/esc/install.sh [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        --silent)
+            SILENT="--silent"
+            ;;
+     esac
+     shift
+done
+
+if [ -z "${VERSION}" ]; then
+    # TODO: remove hardcoded version
+    VERSION=0.2.0
```
Comment (justinvp) [inline dist/esc/install.sh:80]: Is the version going to be 0.5.0?

File: infrastructure/index.ts [hunk 4]
```diff
@@ -308,6 +317,23 @@ for (let entry of fs.readdirSync(distRoot)) {
     }
 }
 
+// Upload all the files in ../dist/esc. We force the Content-Type header to text/plain, so it renders nicely in a web
+// browser when you view the page directly (for example, to inspect the script).
+const escRoot = path.join("..", "dist", "esc");
```
Comment (justinvp) [inline infrastructure/index.ts:322]: Nit

```suggestion
const escRoot = path.join(distRoot, "esc");
```

**Review by justinvp — APPROVED** [review]
LGTM

File: dist/esc/install.ps1 (hunk 1 above)
Comment (lblackstone) [reply dist/esc/install.ps1:14]: It is, but I'll test with 0.2.0 for now, and then follow up with another PR once we cut 0.5.0

**Pull request merged**